btax/front_end_util.py

```diff
@@ -0,0 +1,223 @@
+                else:
+                    extra_row = None
+                row_key = no_spaces_str_equals(keys, row_label)
+                assert row_key == row_label, repr((row_key, row_label))
+                # Logic to find the group that a row
+                # falls into and display the weighted
+                # mean of that row as a bold row summary.
+                # If row['label'] == row['major_grouping'],
+                # then it is a summary row, otherwise
+                # the label differs from the grouping.
+                group = group_data[row_key]
+                row = {
+                    'label': row_label,
+                    'cells': [],
+                    'major_grouping': group['major_grouping'],
+                    'summary_c': '{:.03f}'.format(group['summary_c']),
+                    'summary_nc': '{:.03f}'.format(group['summary_nc']),
+                }
+
+                for col_key in range(0, col_count):
+                    this_row = [r for r in table_data
+                                if row_key == r[0]]
+                    if len(this_row) != 1:
+                        raise ValueError('Failed to find {} ({})'.format(row_key, this_row))
+                    this_row = this_row[0]
+                    value = this_row[col_key + 1]
+                    if do_assertions:
+                        # Do assertions to make sure
+                        # the weighted mean rows, after
+                        # all this reshaping, end up
+                        # being in the right range
+                        key1 = row['major_grouping']
+                        key2 = row['major_grouping'] == row['label']
+                        key3 = col_key + 1
+                        if key3 not in stats[key1][key2]:
+                            minn, maxx = stats[key1][key2][key3] = [None, None]
+                        if minn is None or minn > value:
+                            minn = value
+                        if maxx is None or maxx < value:
+                            maxx = value
+                        stats[key1][key2][key3] = [minn, maxx]
+                    cell = {
+                        'format': {
+                            'divisor': table['cols'][col_key]['divisor'],
+                            'decimals': table['cols'][col_key]['decimals'],
+                        },
+                        'value': value,
+                    }
+                    cell['value'] = value
+                    row['cells'].append(cell)
+                table['rows'].append(row)
+                if extra_row:
+                    # If we made a breakline row,
+                    # add it next because we are
+                    # currently processing the preceding
+                    # row
+                    table['rows'].append(extra_row)
+
+            tables[upper_key][table_id] = table
+    if do_assertions:
+        assertions_on_stats(stats)
+    tables['result_years'] = [2015]
```

Comment:
@PeterDSteinberg, I think it'd be fine to pass that start_year variable here.  Note that I've had PR #107 outstanding for a while.  This would allow one to get different results if a different start year is specified.  I haven't merged it in because we decided not to add that capability to the web app just yet, so it didn't seem necessary.